Feature/m1/client/create todos

Atsuki02 opened the pull request:
## Describe your changes

- Adjust header 

> Make the logo bigger
> Add animation
> Change the nav component form button to link

- Create todos page


## Issue

#18 

## Note

You can't see the tasks in the screen since the seed data doesn't have tasks for today and coming week,
so if you want to see it, please add tasks using createTask api in postman.

## URL

[http://localhost:3000/](http://localhost:3000/)

## Screenshot (if you made the UI)



https://github.com/HAK2024/fairy-share/assets/132407687/4ee82672-6af2-4d0e-84a9-8ca06179173a


https://github.com/HAK2024/fairy-share/assets/132407687/03582729-f096-4bcf-ad4f-2b7632e58a08


**If the user has no tasks**
<img width="946" alt="image" src="https://github.com/HAK2024/fairy-share/assets/132407687/b931ae0d-ff71-441d-9d9d-e4e9bd27feae">

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings and no errors

kanatagu commented on client/app/(todos)/hooks/useUpdateStatusMutation.ts:
```diff
@@ -0,0 +1,22 @@
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ['todos', houseId] })
+    },
+    onError: (error) => {
+      console.error('Update failed', error)
+    },
```
Show toast? We need to handle mutation errors (This is different from query handling, I will do query handling later)

Atsuki02 replied:
I added toast! I'm not sure about mutation errors you're talking about, so let me follow it when you do it!
https://github.com/HAK2024/fairy-share/pull/71/commits/3160c1d542a54421905181f9b034656dc1fcc526

kanatagu commented on client/app/_components/layout/header/Header.tsx:
```diff
@@ -16,41 +15,42 @@ const Header = ({ hasNavigation = true }: HeaderProps) => {
             src={`/images/logo/logo.png`}
             alt={'logo'}
-            width={'150'}
-            height={'150'}
-            className='h-auto w-auto'
+            //  Set the exact ratio of the original image so that you can avoid getting a warn
```
Thanks, you can delete this comment later!

Atsuki02 replied:
https://github.com/HAK2024/fairy-share/pull/71/commits/3d2546c009f6cb80f5d1a9b4eef5815d2280c4ed

kanatagu commented on client/app/page.tsx:
```diff
@@ -1,20 +1,41 @@
+    return <p>Failed to load data.</p>
+  }
+
+  if (!todos) {
```
Because there is no houseId, the `useGetTodoQuery` does not run, right? It means this would never show! You also need to wrap with div container!

Atsuki02 replied:
`useGetTodoQuery` doesn't run when there is no houseId, but it means todos become undefined, so in here I'm showing NoHouseAlert component when todos is undefined, which means the user is not in any house

I thought it's would be a good idea, but when the user has no house, they don't necessarily have to create a house and they can just wait for an invitation, so I think showing NoHouseAlert makes more sense.

Atsuki02 replied:
I deleted the Alert component!
https://github.com/HAK2024/fairy-share/pull/71/commits/38d451546e7f85f67065d9ffda6c7ee1c0ea268b

kanatagu commented on client/app/page.tsx:
```diff
@@ -1,20 +1,41 @@
+    <div className='flex flex-col gap-8 px-4 pb-10 pt-8 text-slate-800 md:px-14 md:pb-20 md:pt-10 '>
       <Heading
-        title='Fairy share'
+        title={todos.houseName}
```
Can you also change the rules button to redirect to `/rules`?

Atsuki02 replied:
Fixed it!
https://github.com/HAK2024/fairy-share/pull/71/commits/4b1deb9c93b0fc0758b4f0284dc44290cb814eec

kanatagu commented on client/app/page.tsx:
```diff
@@ -1,20 +1,41 @@
 
   return (
-    <div className='px-4 pb-10 pt-8 md:px-14 md:pb-20 md:pt-10'>
+    <div className='flex flex-col gap-8 px-4 pb-10 pt-8 text-slate-800 md:px-14 md:pb-20 md:pt-10 '>
```
You can actually not scroll and not see the bottom element because of the footer...

<img width="412" alt="Screenshot 2024-03-13 at 4 14 09 PM" src="https://github.com/HAK2024/fairy-share/assets/66394413/93afb91d-742f-48e7-af66-858a5f794c90">

Atsuki02 replied:
I fixed it by adding margin to layout since footer doesn't have height!
https://github.com/HAK2024/fairy-share/pull/71/commits/fca9b3a369c8a2f831592cc8f09c6f4529a69da8

Atsuki02 replied:
Sorry this change affects the auth page, so I'll change it

kanatagu commented on client/app/(todos)/components/TaskItem.tsx:
```diff
@@ -0,0 +1,26 @@
+      <p className='max-w-64 truncate text-lg font-semibold md:max-w-36 lg:max-w-72'>
+        {task.title}
+      </p>
+      <Checkbox
+        className='h-6 w-6 md:h-8 md:w-8'
+        checked={task.isCompleted}
+        onClick={handleUpdateTaskStatus}
+      />
```
Can you add disabled while the mutation is running? You don't want a user to push this button many times before it finishes

Atsuki02 replied:
Fixed it!
https://github.com/HAK2024/fairy-share/pull/71/commits/7821e941b5e5a8dd81164ab6149b427502f0eb01

kanatagu commented on client/app/(todos)/components/Alert.tsx:
```diff
@@ -0,0 +1,33 @@
+  linkText,
+  additionalText = '',
+}: AlertProps) => (
+  <div className='flex flex-row gap-6 rounded-sm border-[1px] border-red-400 bg-red-100 px-5 py-6 md:px-8'>
```
I think for smartphones, it is too much gap. `gap-3` for sp?

Atsuki02 replied:
Fixed it!
https://github.com/HAK2024/fairy-share/pull/71/commits/3fb6fce2e7e765f07138238ee4aaf574d968c759
https://github.com/HAK2024/fairy-share/pull/71/commits/2398a2838cf9a13e1574f9df78857418e63208c8

Atsuki02 commented on client/app/(auth)/api/registerApi.ts:
```diff
@@ -7,5 +7,5 @@ export const registerApi = async (data: RegisterSchema) => {
     data,
   )
-  return response.data
+  return response
```
Oh sorry, it was unintentional change
Fixed it!
https://github.com/HAK2024/fairy-share/pull/71/commits/fce9d7ea461b066059d1905fee035ad15582d128

Atsuki02 commented on client/app/(todos)/components/TaskItem.tsx:
```diff
@@ -0,0 +1,26 @@
+
+  return (
+    <li className='flex items-center justify-between gap-6 border-b-[1px] border-slate-400 pb-1'>
+      <p className='max-w-64 truncate text-lg font-semibold md:max-w-36 lg:max-w-72'>
```
Oh that's true!
I fixed it
https://github.com/HAK2024/fairy-share/pull/71/commits/6fee9bdd89ebe9028c69dde586afb7e4599dd672

Atsuki02 commented on client/app/(todos)/components/TodosList.tsx:
```diff
@@ -0,0 +1,25 @@
+        <TaskList title='Today' tasks={todos.todayTasks} />
+        <TaskList title='Coming this week' tasks={todos.weekTasks} />
+      </div>
+      {!todos.hasUnpaidPayments && (
```
Oh yeah! I forgot to get it back when I tested to show it!
https://github.com/HAK2024/fairy-share/pull/71/commits/02402103c1b3e702fcd55ae4c6329d8e111bdd1e

kanatagu reviewed (approved):
LGTM!

You can merge this branch now but I think I found a bug in server side. When you make tomorrow's task, you don't get it as coming this week task. Tomorrow's task should include in coming this week. Am I correct? 

<img width="1104" alt="Screenshot 2024-03-14 at 2 26 52 PM" src="https://github.com/HAK2024/fairy-share/assets/66394413/2d43b2cf-6541-4c6c-ba59-4c33941c6514">


<img width="838" alt="Screenshot 2024-03-14 at 2 27 08 PM" src="https://github.com/HAK2024/fairy-share/assets/66394413/a757a37a-39d7-4d68-820b-4cd623808f80">

Atsuki02 commented:
> LGTM!
> 
> You can merge this branch now but I think I found a bug in server side. When you make tomorrow's task, you don't get it as coming this week task. Tomorrow's task should include in coming this week. Am I correct?
> 
> <img alt="Screenshot 2024-03-14 at 2 26 52 PM" width="1104" src="https://private-user-images.githubusercontent.com/66394413/313001572-2d43b2cf-6541-4c6c-ba59-4c33941c6514.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA0NTU1ODcsIm5iZiI6MTcxMDQ1NTI4NywicGF0aCI6Ii82NjM5NDQxMy8zMTMwMDE1NzItMmQ0M2IyY2YtNjU0MS00YzZjLWJhNTktNGMzMzk0MWM2NTE0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE0VDIyMjgwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJmOGJhY2RhOTI5MDE3MzBiZDc4YmU3MGJjNDYyZTM4NmQ2NzJiNjA5NDIxNmVhMWEwYTc4ZWNkYmQ4OTYyY2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-IUL_akhMpuK6jbAJt1hX6zhC3XzoKZt8TiyCBsr7_M"> <img alt="Screenshot 2024-03-14 at 2 27 08 PM" width="838" src="https://private-user-images.githubusercontent.com/66394413/313001563-a757a37a-39d7-4d68-820b-4cd623808f80.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA0NTU1ODcsIm5iZiI6MTcxMDQ1NTI4NywicGF0aCI6Ii82NjM5NDQxMy8zMTMwMDE1NjMtYTc1N2EzN2EtMzlkNy00ZDY4LTgyMGItNGNkNjIzODA4ZjgwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE0VDIyMjgwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAxYWJjM2MzZTI5NzAxZmQ1YzM3MzljZTc5ZTFhNjhhOGRjYWQ3YmRlZGU5YTE0YmJlMTQ3YmRhZjdkMGIxZjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.wV0MxF0oEZ-bBQ1ivzIJ0a3e9JdQvsRhioUFc0vzPwo">



I fixed the bug! It happens becuase I was using the same variable for weeksTasks as todayTasks and when I mutate currentDate in todayTasks, it affected weekTasks. So I fixed it by seperating the variables.
https://github.com/HAK2024/fairy-share/pull/71/commits/fd4e4702ff26f70fa78929ef75368bbdcef047f9
